File: omp/src20/cell_metrics.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>
#include <errno.h>
#include <math.h>

#include "struct_ext.h"
#include "pcg_ext.h"
#include "cell_metrics.h"
#include "allocate.h"

extern int
CELL_METRICS(void)
{
	double V0, RV0;
	int i;

/************
 * ALLOCATE *
 ************/
	VOLCEL = (double *)allocate_vector(sizeof(double),ICELTOT);
	RVC    = (double *)allocate_vector(sizeof(double),ICELTOT);

/*********************************
 * VOLUME, AREA, PROJECTION etc. *
 *********************************/
	XAREA = DY * DZ;
	YAREA = DZ * DX;
	ZAREA = DX * DY;

	RDX = 1.0 / DX;
	RDY = 1.0 / DY;
	RDZ = 1.0 / DZ;

	RDX2 = 1.0 / (pow(DX, 2.0));
	RDY2 = 1.0 / (pow(DY, 2.0));
	RDZ2 = 1.0 / (pow(DZ, 2.0));

	R2DX = 1.0 / (0.5 * DX);
	R2DY = 1.0 / (0.5 * DY);
	R2DZ = 1.0 / (0.5 * DZ);

	V0  = DX * DY * DZ;
	RV0 = 1.0 / V0;

	for(i=0; i<ICELTOT; i++) {
		VOLCEL[i] = V0;
		RVC[i]    = RV0;
	}

	return 0;
}

File: multicore-f/L1/src/cell_metrics.f
!C
!C***  
!C*** CELL_METRICS
!C***      
!C
      subroutine CELL_METRICS

      use STRUCT
      use PCG

      implicit REAL*8 (A-H,O-Z)

!C
!C-- ALLOCATE
      allocate (VOLCEL(ICELTOT))
      allocate (   RVC(ICELTOT))

!C
!C-- VOLUME, AREA, PROJECTION etc.
      XAREA= DY * DZ
      YAREA= DX * DZ
      ZAREA= DX * DY

      RDX= 1.d0 / DX
      RDY= 1.d0 / DY
      RDZ= 1.d0 / DZ

      RDX2= 1.d0 / (DX**2)
      RDY2= 1.d0 / (DY**2)
      RDZ2= 1.d0 / (DZ**2)

      R2DX= 1.d0 / (0.50d0*DX)
      R2DY= 1.d0 / (0.50d0*DY)
      R2DZ= 1.d0 / (0.50d0*DZ)

       V0= DX * DY * DZ
      RV0= 1.d0/V0

      VOLCEL=  V0
      RVC   = RV0

      return
      end
      
